### PR TITLE
Switch to internal event for Monitor

### PR DESCRIPTION
**Pull request by HaoK**
@ajcvickers @pakrym 

Fixes https://github.com/aspnet/Options/issues/204

### CODE REVIEW

File: src/Microsoft.Extensions.Options/OptionsMonitor.cs [hunk 1]
```diff
@@ -16,7 +16,7 @@ public class OptionsMonitor<TOptions> : IOptionsMonitor<TOptions> where TOptions
         private readonly IOptionsMonitorCache<TOptions> _cache;
         private readonly IOptionsFactory<TOptions> _factory;
         private readonly IEnumerable<IOptionsChangeTokenSource<TOptions>> _sources;
-        private List<Action<TOptions>> _listeners = new List<Action<TOptions>>();
+        internal event EventHandler<ChangeArgs> _onChange;
```
Comment (pakrym) [inline src/Microsoft.Extensions.Options/OptionsMonitor.cs]: Event can be `Action` too

**Comment by HaoK** [comment]
Updated to use `Action<TOptions>` which cleans things up a bit

File: src/Microsoft.Extensions.Options/OptionsMonitor.cs [hunk 2]
```diff
@@ -72,26 +72,32 @@ public virtual TOptions Get(string name)
...
         public IDisposable OnChange(Action<TOptions> listener)
         {
-            var disposable = new ChangeTrackerDisposable(_listeners, listener);
-            _listeners.Add(listener);
+            var disposable = new ChangeTrackerDisposable(this, listener);
+            _onChange += disposable.OnChange;
             return disposable;
         }
 
+        internal class ChangeArgs : EventArgs
```
Comment (pakrym) [inline src/Microsoft.Extensions.Options/OptionsMonitor.cs]: Don't need anymore

**Comment by HaoK** [comment]
ff4200632970c53a3a6c860219d10477d74060d6

**Pull request closed without merging**